src/main/java/commons/header/Properties.java

```diff
@@ -0,0 +1,45 @@
+package commons.header;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Properties {
+    private Map<String, String> properties;
+
+    // Default constructor that initializes an empty properties map
+    public Properties() {
+        this.properties = new HashMap<>();
+    }
+    // Constructor that initializes the properties with a deep copy of the provided map
+    public Properties(Map<String, String> properties) {
+        this.properties = new HashMap<>(properties);
+    }
+
+    public void setProperty(String key, String value) {
+        properties.put(key, value);
+    }
+
+    public String getProperty(String key) {
+        return properties.get(key);
+
+    }
+    public void removeProperty(String key) {
+        properties.remove(key);
+    }
+
+    public boolean containsKey(String key) {
+        return properties.containsKey(key);
+    }
```

Comment:
minor nitpick but I think it would be better to name this as `containsProperty` just to be more specific